### DISCUSSION TITLE
Adding Github Achievements

### DISCUSSION THREAD

**Discussion by fundor333** [lowlighter/metrics#1090] [Ideas]
Can we have the [Github Achievements](https://github.blog/2022-06-09-introducing-achievements-recognizing-the-many-stages-of-a-developers-coding-journey/) added into the metrics?

**Comment by lowlighter** [comment]
Hi 👋 !

Can you give me a bit more informations about what you expect ?

Would it be like a duplicate section of the current GitHub achievements ? Or justs stats that would display the number of achievements earned (with possibly the rank of them) ?

Also note that there are no GitHub API for them (and from what was posted on github.blog and the facts that the list was removed from documentation I suspect there won't be any) so some stuff may not be doable depending on what it is 😅

**Reply by jayvdb** [reply]
Worst case, the HTML could be scraped and parsed.

Could be useful for profiles rendered on https://metrics.lecoq.io/ or self-hosted.  Doesn't seem so useful when rendered on Github itself.  One benefit of a plugin would be to create a personal archive of the achievements they hand out.

But then, what to do with it?  Unless it is presented as-is with the popup explanation and 'proof', many of the badges have lost their meaning.

And some are easter-eggs at best. e.g. YOLO isnt well understood per https://github.com/github-community/community/discussions/18159

And then there is intellectual property rights. The decals have a copyright that would need to be explored.  They are not mentioned on https://docs.github.com/en/site-policy/github-terms/github-terms-of-service#1-githubs-rights-to-content , but that does say 
> The look and feel of the Website and Service is copyright © GitHub, Inc. All rights reserved. You may not duplicate, copy, or reuse any portion of the HTML/CSS, Javascript, or visual design elements or concepts without express written permission from GitHub.

So probably not openly licensed; even if they are openly licensed, any rendering of them likely needs to include attribution.

Also there are database rights for the actual data, if this profile generator is using their generated data instead of determining the contents from scratch.  Database rights havent been accepted in the US, so this may be a non-issue as it seems GitHub is using US-law per https://docs.github.com/en/site-policy/github-terms/github-terms-of-service#1-governing-law

So, the badges probably cant be displayed as-is unless permission is obtained - the badges would need to be given a new UI, which IMO makes the duplication more confusing and reduces the benefits.

**Reply by lowlighter** [reply]
Yeah these are good points.

Unless these are included in primer css, I don't think we can use them as is indeed (at least on a web instance)

If it were produced in a SVG by scraping stuff, I'm not sure how it fall legally. Technically the action don't include copyrighted content but the resulting image does. Not sure if the responsability is on the user/image owner or the tool itself 

This GitHub feature is still in beta and I feel like there are a lot of mixed feelings on it. So not sure if it'll impact a lot the final product

**Comment by jayvdb** [comment]
Possibly more useful to consider is whether any of the GitHub Achievements highlight any useful features missing from this project.

Full list and some notes at https://github.com/drknzz/GitHub-Achievements

The "Arctic Code Vault" and "Mars 2020" repository list might be useful for the notable plugin, indicating "very" notable repos.

The "Starstruck" achievement may already be covered by one of the achievement plugins badges...?
However the list of notable self-created repos which powers "Starstruck" seems to be a missing feature, and one which is quite important IMO.  It seems 16 stars is the threshold.  I notice that https://gist.githubusercontent.com/lowlighter/3c6eaedf50273adfb7a510822672f570/raw/d55bf1bacbb3c1a2aa89fd4119a09179597e4ebf/general.svg "notable" doesn't include `@lowlighter/metrics`, which seems a bit odd.  My own repo with 16 stars is also missing from notable repos.  I dont see any boolean flag for the notable plugin to exclude self-created repositories, so I am wondering why they are being excluded.  A valid reason is usually self-created repos are very likely to be not-notable, but 16 stars would be a good default for a threshold.  Always game-able , but IMO that is self-defeating - I wouldn't want one of my repos showing up on the profile as notable unless it honestly is. (Ahh, I see we've discussed this before at https://github.com/lowlighter/metrics/discussions/654#discussioncomment-1654564 )

The Pair Extraordinaire badge is IMO useless, as the co-author on commits is usually a person (or bot) doing a squash+merge.  Sometimes it is someone offering a ["suggestion"](http://haacked.com/archive/2019/06/03/suggested-changes/) which gets added, but that usually isnt real pair programming - more often than not it is fixing spelling or other simple one-line fixups because multi-line suggestions using the github suggestion tool is clunky.

The Quickdraw & YOLO badges are a bit cheeky, and not yet well understood.  I am expecting there are more criteria than are publicly known, as I don't seem to have Quickdraw yet I know I have done lots of immediate merges over the years.

IMO "Pull Shark" badge is already well covered by existing plugins, esp the notable and achievement plugins, but even base.

"Galaxy Brain" about discussions seems to be mostly covered by existing plugins.

**Reply by lowlighter** [reply]
Yeah the notable plugin was historically intended to only display organizations in which an user contributed, but then it was extended to support user repositories except self-owned, before getting additional features like it is now

The constraint was never removed but I kind of forgot how it is done lol😅 
Will evetually be done by #1143 👍 

As for other achievements, I guess it's not necessary to replicate what is already done by GitHub (personally I think some of them are kind of weird and apart for a few one, they don't tell much about a user)
